pandas/core/generic.py

```diff
@@ -1154,29 +1154,32 @@ def rename_axis(self, mapper, axis=0, copy=True, inplace=False):
         --------
         **Series**
 
-        >>> s = pd.Series([1, 2, 3])
-        >>> s.rename_axis("foo")
-        foo
-        0    1
-        1    2
-        2    3
-        dtype: int64
+        >>> s = pd.Series(["dog", "cat", "monkey"])
+        >>> s.rename_axis("animal")
+        animal
+        0    dog
+        1    cat
+        2    monkey
+        dtype: object
 
         **DataFrame**
 
-        >>> df = pd.DataFrame({"A": [1, 2, 3], "B": [4, 5, 6]})
-        >>> df.rename_axis("foo")
-             A  B
-        foo
-        0    1  4
-        1    2  5
-        2    3  6
-
-        >>> df.rename_axis("bar", axis="columns")
-        bar  A  B
-        0    1  4
-        1    2  5
-        2    3  6
+        >>> df = pd.DataFrame({"name": ["dog", "cat", "monkey"], "legs": [4, 4, 2]})
+             name  legs
+        0     dog     4
+        1     cat     4
+        2  monkey     2
+        >>> df.rename_axis("animal")
+                  name  legs
+        animal
+        0          dog     4
+        1          cat     4
+        2       monkey     2
+        >>> df.rename_axis("id", axis="columns")
```

Comment:
In this case, the concept `id` refers to `name` and `legs`, which does not make sense.

It could make more sense to have the name as the index, and then may be a column `num_legs` and a column `num_arms` and call the axis `limbs`.

Reply:
Makes sense